### PR TITLE
fix: redo MP drawer layout changes with mobile fixed

### PR DESCRIPTION
**Pull request by just-toby**
## Description

this is the same change as #6260  but with the mobile behavior fixed, and an extra e2e test added. the original PR was completely missing the mobile changes we need.

[context here](https://uniswapteam.slack.com/archives/C02T729PMQE/p1680214677184219)

## Screen Capture
| Before           | After           |
| ---------------- |-----------------|
|  

https://user-images.githubusercontent.com/66155195/228985257-fbc55321-4312-4c37-8838-5c08a40770ed.MP4 

| 

https://user-images.githubusercontent.com/66155195/228985605-07a0d12c-8816-4aea-bbb6-f6afc797165b.mov 

|

#### Manual
manually verified the mobile behavior this time, in addition to the regular desktop behavior.

added an e2e test to verify that you can close the bottom sheet by tapping on the scrim on mobile

#### Automated
- [ ] Unit test
- [x] Integration/E2E test

### CODE REVIEW

File: src/components/WalletDropdown/index.tsx [hunk 1]
```diff
@@ -206,7 +212,7 @@ function WalletDropdown() {
           <DefaultMenu />
         </WalletDropdownScrollWrapper>
       </WalletDropdownWrapper>
-    </>
+    </Wrapper>
```
Comment (cartcrom) [inline src/components/WalletDropdown/index.tsx:215]: w/ adding a new level of wrapping, I think we could improve our naming convention here, potentially:
```
<Container>
    <Wrapper>
        <ScrollBoundary ref={scrollRef}>
        </ScrollBoundary>
   </Wrapper>
</Container>
```

**Review by cartcrom — CHANGES_REQUESTED** [review]
I'm not able to click anything under DRAWER_MARGIN it seems while on mobile

https://user-images.githubusercontent.com/39385577/229880815-bf807887-c021-4aba-ad4f-227adeb86522.mov

**Comment by just-toby** [comment]
@cartcrom thanks for the check - i've added another e2e test for mobile navigation to catch that case too

**Review by cartcrom — APPROVED** [review]
LGTM, thx for adding tests!

File: cypress/e2e/landing.test.ts [hunk 2]
```diff
@@ -26,7 +26,15 @@ describe('Landing Page', () => {
 
   it('allows navigation to pool', () => {
     cy.viewport(2000, 1600)
+    cy.visit('/swap')
     cy.get(getTestSelector('pool-nav-link')).first().click()
     cy.url().should('include', '/pools')
   })
+
+  it('allows navigation to pool on mobile', () => {
+    cy.viewport(375, 667)
```
Comment (zzmp) [inline cypress/e2e/landing.test.ts:35]: Where are these numbers from? Can you leave a comment, and maybe make them a constant?

```
const DESKTOP_DIMENSIONS = [2000, 1600]
const MOBILE_DIMENSIONS = [375, 667]
...
cy.viewport(...DESKTOP_DIMENSIONS)
...
cy.viewport(...MOBILE_DIMENSIONS)
```

File: cypress/e2e/wallet-dropdown.test.ts [hunk 3]
```diff
@@ -77,4 +77,12 @@ describe('Wallet Dropdown', () => {
     cy.get(getTestSelector('theme-auto')).click()
     cy.get(getTestSelector('wallet-header')).should('have.css', 'color', 'rgb(119, 128, 160)')
   })
+
+  it('should use a bottom sheet and dismiss when on a mobile screen size', () => {
+    visit(true)
+    cy.viewport('iphone-6')
```
Comment (zzmp) [inline cypress/e2e/wallet-dropdown.test.ts:83]: why not use an alias like this in landing.test.ts? Then you wouldn't need *_DIMENSIONS.

File: src/components/WalletDropdown/index.tsx [hunk 4]
```diff
@@ -76,32 +76,45 @@ const WalletDropdownScrollWrapper = styled.div`
...
-    height: calc(100% - ${DRAWER_TOP_MARGIN_MOBILE_WEB});
     border-bottom-right-radius: 0px;
     border-bottom-left-radius: 0px;
     box-shadow: unset;
+    transition: top ${({ theme }) => theme.transition.duration.medium};
   }
 
   @media screen and (min-width: 1440px) {
-    right: ${({ open }) => (open ? DRAWER_MARGIN : '-' + DRAWER_WIDTH_XL)};
+    margin-right: ${({ open }) => (open ? 0 : '-' + DRAWER_WIDTH_XL)};
```
Comment (zzmp) [inline src/components/WalletDropdown/index.tsx:117]:
```suggestion
    margin-right: ${({ open }) => (open ? 0 : `-${DRAWER_WIDTH_XL}`)};
```

**Review by zzmp — CHANGES_REQUESTED** [review]

**Review by zzmp — APPROVED** [review]

**Pull request merged**